ui/shared/AddressInput.qml

```diff
@@ -31,13 +33,33 @@ Item {
     function validate(inputValue) {
         if (!inputValue) inputValue = selectedAddress
         let isValid =
-            (inputValue && inputValue.startsWith("0x") && Utils.isValidAddress(inputValue)) ||
-            isValidEns(inputValue)
+            (inputValue && 
+              (inputValue.startsWith("0x") && Utils.isValidAddress(inputValue) || isValidEns(inputValue)) ||
+              inputValue.startsWith("@"))
         inpAddress.validationError = isValid ? "" : validationError
         root.isValid = isValid
         return isValid
     }
 
+    property var validateAsync: Backpressure.debounce(inpAddress, 400, function (inputValue) {
+        root.isPending = true
+        var name = inputValue.startsWith("@") ? inputValue.substring(1) : inputValue
+        walletModel.resolveENS(name)
+    });
```

Comment:
For the very same reason as stated above, ENS resolution is baked hard into the component, as opposed to having the validation handler passed down from the outside world. Only downside (as usual) is that `AddressInput` is now dependent on `walletModel`.

I think we need to find a way to have our UI components provide certain logic, without them being dependent on application specific models.

Something to think about in the future @emizzle

Reply:
"Tricky" part here though is that we rely on `Connections` and signals coming from Nim (e.g. `ensWasResolved`) which we can't make super configurable (I believe). So that would mean the `Connections` would need to be moved outside of the `AddressInput`, which in turn defeats the purpose of it being able to asynchronously validate...

Reply:
Actually, both could be passed outside of the component. `AddressInput` can get a signal that can be triggered when the `walletModel` signal is triggered. There is an API that could be of use here: 

`AddressInput.qml`
```
Item {
    signal ensWasResolved(string address)
}
```

parent of `AddressInput`:
```
AddressInput {
    resolveEns: walletModel.resolveEns
    Component.onCompleted: {
        walletModel.onEnsWasResolved.connect(ensWasResolved)
    }
}
```

There's a *slight* possibility the signal from `walletModel` can be passed directly to the property, but i'm not 100% sure, ie:
```
AddressInput {
    resolveEns: walletModel.resolveEns
    ensWasResolved: walletModel.onEnsWasResolved
}
```

Reply:
So if we make `walletModel` APIs the default anyways and given that the chances are almost 0% that someone would pass a different functionality here, I think we can then also just go with baking walletModel APIs in (for now) until we found a solution where the QML components can come with corresponding Nim components.